### PR TITLE
Records (Python)

### PR DESCRIPTION
**Pull request by SheepTester**
Per surveys, records will use colons, newlines/semicolons, and curly brackets

### CODE REVIEW

File: python/syntax.lark [hunk 1]
```diff
@@ -94,12 +97,13 @@ value: NUMBER
...
      | listval
+     | recordval
 
 //constants
 BOOLEAN: ("true" | "false")
 COMMENT: "//" /[^\n]/*
-OR: "||"
-AND: "&&"
+OR: ("||" | "|")
+AND: ("&&" | "&")
```
Comment (Ashvin-Ranjan) [inline python/syntax.lark:106]: Do we want to make | and || different?

File: python/scope.py [hunk 2]
```diff
@@ -517,8 +534,10 @@ def type_check_expr(self, expr):
 			for i, e in enumerate(rest):
 				if e != first:
 					self.errors.append(TypeCheckError(expr.children[i+1], "The list item #%s's type is %s while the first item's type is %s" % (i + 2, e, first)))
-			
+
```
Comment (Ashvin-Ranjan) [inline python/scope.py:537]: very cool github

File: python/run.n [hunk 3]
```diff
@@ -51,9 +51,16 @@ let ch:char = \n
...
 
-var list = [1]
+// var list = [1]
+
+let record = {
+  ch
+  anyways: main; wo2w: true
+}
+print(record)
+//print ({ nested: [(record, 3)] } + 1)
```
Comment (Ashvin-Ranjan) [inline python/run.n:63]: what is ch?

File: python/syntax.lark (hunk 1 above)
Comment (SheepTester) [reply python/syntax.lark:106]: in the JS branch they're the same; "and" either becomes bitwise and for ints and doesn't short circuit or becomes logical and for bools and does short circuit

File: python/scope.py (hunk 2 above)
Comment (SheepTester) [reply python/scope.py:537]: lol my editor is set to automatically remove trailing whitespace

File: python/run.n (hunk 3 above)
Comment (SheepTester) [reply python/run.n:63]: Above, `let ch:char = \n` I believe

Comment (Ashvin-Ranjan) [reply python/run.n:63]: oh ok, but why does it not have a key?

File: python/syntax.lark (hunk 1 above)
Comment (Ashvin-Ranjan) [reply python/syntax.lark:106]: ok epic

File: python/run.n (hunk 3 above)
Comment (SheepTester) [reply python/run.n:63]: `ch` on its own is implicitly a shorthand for `ch: ch`

**Pull request merged**